backwards-compat-tests/scripts/utils.sh

```diff
@@ -129,37 +129,47 @@ version_lt() {
 
 ################################### Entry Points
 
-# Get $OLD_VERSION and $NEW_VERSION for Risingwave
-get_rw_versions() {
-  # For backwards compat test we assume we are testing the latest version of RW (i.e. latest main commit)
-  # against the Nth latest release candidate, where N > 1. N can be larger,
-  # in case some old cluster did not upgrade.
-  local VERSION_OFFSET=4
-
-  # First we obtain a list of versions from git branch names.
-  # Then we normalize them to semver format (MAJOR.MINOR.PATCH).
-  echo "--- git branch origin output"
-  git branch -r | grep origin
-
-  # Extract X.Y.Z tags
-  echo "--- VERSION BRANCHES"
-  local tags=$(git tag | grep -E "^v[0-9]+\.[0-9]+\.[0-9]+$" | tr -d 'v' | tr -d ' ')
-  echo "$tags"
-
-  # Then we sort them in descending order.
-  echo "--- VERSIONS"
-  local sorted_versions=$(echo -e "$tags" | sort -t '.' -n)
-  echo "$sorted_versions"
-
-  # Then we take the Nth latest version.
-  # We set $OLD_VERSION to this.
-  OLD_VERSION=$(echo -e "$sorted_versions" | tail -n $VERSION_OFFSET | head -1)
+get_old_version() {
+   # For backwards compat test we assume we are testing the latest version of RW (i.e. latest main commit)
+   # against the Nth latest release candidate, where N > 1. N can be larger,
+   # in case some old cluster did not upgrade.
+   if [[ -z $VERSION_OFFSET ]]
+   then
+       local VERSION_OFFSET=4
+   fi
+
+   # First we obtain a list of versions from git branch names.
+   # Then we normalize them to semver format (MAJOR.MINOR.PATCH).
+   echo "--- git branch origin output"
+   git branch -r | grep origin
+
+   # Extract X.Y.Z tags
+   echo "--- VERSION BRANCHES"
+   local tags=$(git tag | grep -E "^v[0-9]+\.[0-9]+\.[0-9]+$" | tr -d 'v' | tr -d ' ')
+   echo "$tags"
+
+   # Then we sort them in descending order.
+   echo "--- VERSIONS"
+   local sorted_versions=$(echo -e "$tags" | sort -t '.' -n)
+   echo "$sorted_versions"
```

Comment:
If we finally have many patch versions for the 2nd latest minor version, seems we then won't test for the 3rd latest minor version and so on? Like `1.5.0` `1.4.3` `1.4.2` `1.4.1`.

What about test for latest patch versions for the most recent 4 minor versions?

Reply:
Makes sense, although we don't really release patch versions much. I will track it in a separate issue.

Reply:
https://github.com/risingwavelabs/risingwave/issues/13913